packages/expo-image-picker/android/src/main/java/expo/modules/imagepicker/ImagePickerModule.kt

```diff
@@ -221,7 +245,58 @@ class ImagePickerModule(private val mContext: Context,
       setOutputCompressQuality(pickerOptions.quality)
     }
 
-    startActivityOnResult(cropImageBuilder.getIntent(context), CropImage.CROP_IMAGE_ACTIVITY_REQUEST_CODE, promise)
+    startActivityOnResult(cropImageBuilder.getIntent(context), CropImage.CROP_IMAGE_ACTIVITY_REQUEST_CODE, promise, pickerOptions)
+  }
+
+  //endregion
+
+  // region ActivityEventListener
+
+  override fun onNewIntent(intent: Intent) = Unit
+
+  override fun onActivityResult(activity: Activity, requestCode: Int, resultCode: Int, data: Intent?) {
+    if (shouldHandleOnActivityResult(activity, requestCode)) {
+      mUIManager.unregisterActivityEventListener(this)
+
+      val promise = if (mWasDestroyed) {
+        PendingPromise(pickerResultStore)
+      } else {
+        mPromise!!
+      }
+      val pickerOptions = mPickerOptions!!
+
+      mPromise = null
+      mPickerOptions = null
+
+      handleOnActivityResult(promise, activity, requestCode, resultCode, data, pickerOptions)
+    }
+  }
+
+  //endregion
+
+  //region activity for result
```

Comment:
Same here, is it possible to delegate this region to different class.
I'd like to stick more & more to SOLID rules and similar 😉